src/compiler/checker.ts

```diff
@@ -2895,6 +2902,34 @@ namespace ts {
             return unknownType;
         }
 
+        function getTypeOfBasePropertyDeclaration(declaration: VariableLikeDeclaration) {
+            if (declaration.parent.kind === SyntaxKind.ClassDeclaration) {
+                const property = getPropertyFromBaseInterfaces(<ClassLikeDeclaration>declaration.parent, declaration.symbol.name);
+                if (property) {
+                    return getTypeOfSymbol(property);
+                }
+            }
+        }
+
+        function getPropertyFromBaseInterfaces(declaration: ClassLikeDeclaration, propertyName: string): Symbol {

```

Comment:
I thought that a derived class property would override the base class property (making their types unrelated), but on playing with it in the playground, it looks like that's not the case -- it has to be a subclass of the base class' property's type. I'll add code to check the base class too.